Add rename script

seothemes opened the pull request:
I've created a node package to rename strings in the theme based on the theme claim package mentioned [here](https://github.com/studiopress/genesis-sample/issues/142#issuecomment-455137715).

**Installation:**

Save as a dev dependency:

```shell
npm install genesis-theme-claim@0.1.1 --save-dev
```

**Usage:**

Run the following command and follow the CLI instructions:

```shell
npm run rename
```

I've probably missed a few strings but the bulk of it is working.

nickcernis commented:
Thanks for this, @seothemes. I like the interactive approach:

<img width="957" alt="Screen Shot 2019-07-10 at 12 27 52" src="https://user-images.githubusercontent.com/647669/60962273-2e1c7f80-a30e-11e9-9528-415974451e5a.png">

As you predicted, there are still several “sample” strings left after running `npm run rename`.

<img width="760" alt="Screen Shot 2019-07-10 at 12 24 05" src="https://user-images.githubusercontent.com/647669/60962069-c108ea00-a30d-11e9-927e-aab1dc36a1be.png">

Do you think the `theme-claim`/`genesis-theme-claim` packages are flexible enough to allow us to process those too (and potentially reset version numbers and `@since` comments as you suggest)? Or would we be better off writing our own script?

seothemes commented:
@nickcernis it will definitely handle all of the string replacements. I’ll get it updated tomorrow to include those files that it missed.

I’m not sure about version numbers though, maybe it would be better to use something like [bump-regex](https://www.npmjs.com/package/bump-regex) to allow the version to be bumped independently.

nickcernis commented:
>  maybe it would be better to use something like bump-regex to allow the version to be bumped independently.

I was thinking more that, if someone's using Genesis Sample as a new theme for a project, it's undesirable to have all version numbers reading 3.0.1 (instead of 0.1.0 or 1.0.0), and all '@since' documentation comments reading different values that may exceed the project's first version (all existing `@since` should use the first version of the theme, ideally).

seothemes commented:
@nickcernis ok, I've bumped Genesis Theme Claim to version 0.3.0 which replaces all strings *and* resets the version number. Ready for you to take a look.

nickcernis commented:
Nice work, @seothemes! Resetting versions to 1.0.0 is a great touch.

I found a few tiny 'Sample' strings still remaining in file/function names (ignore the `package.json`):

<img width="487" alt="Screen Shot 2019-07-18 at 21 52 32" src="https://user-images.githubusercontent.com/647669/61487530-6f63fd80-a9a6-11e9-94b1-1e7cb7d6292c.png">

Should we delete the genesis-sample.pot file too? (An alternative might be to delete all existing *.pot files in the makepot script before creating a new one, which would recreate it with the correct filename the next time `npm run makepot` is run.)

I suppose someone might expect the changelog to be wiped for a new theme too, but they can update this themselves pretty easily.

Apart from that, it looks great. Thank you again for all the work that went into this.

seothemes commented:
@nickcernis updated to 0.4.0 which adds a new function to rename the POT file and WooCommerce CSS file. I wasn't able to find any "sample" strings after running the latest script.

Have also added a changelog reset function which should reset it to something like this:

```markdown
# My Theme Changelog

## [1.0.0] - (current date)
* Initial commit.
```

Is there anything else you can think of to improve it further?

nickcernis commented:
Great job! The only two things I found with a new theme string of “Genesis NewTheme” were:

1. The WooCommerce script is renamed to `new-theme`:

genesis-new-theme-woocommerce.css

But it's enqueued as (`newtheme`):

`'src'     => get_stylesheet_directory_uri() . '/lib/woocommerce/genesis-newtheme-woocommerce.css',`

(Input string for theme name was `Genesis NewTheme`.)

We should probably just change the src to use `genesis_get_theme_handle()`

`'src'     => get_stylesheet_directory_uri() . '/lib/woocommerce/genesis-' . genesis_get_theme_handle() . '-woocommerce.css',`
de

2. The pot file is renamed to `genesis-newtheme.pot`, which should be `genesis-new-theme.pot`, which is what you'll see created if you run `npm run makepot`.

Both are probably edge cases to do with the “NewTheme” string.

I'm otherwise happy to merge this pending approval from one more reviewer.

seothemes commented:
Would it be better to just rename the stylesheet to `woocommerce.css`? Having the theme name there is not really necessary since it's already in the handle. Can add to this PR.

For the POT file, that is expected. To have `genesis-new-theme.pot` you would need to enter "Genesis New Theme" as the theme name. This will need to be documented somewhere. Should I add some basic info to the readme?

nickcernis commented:
> Would it be better to just rename the stylesheet to woocommerce.css? Having the theme name there is not really necessary since it's already in the handle. Can add to this PR.

Good idea.

> For the POT file, that is expected. To have genesis-new-theme.pot you would need to enter "Genesis New Theme" as the theme name. This will need to be documented somewhere. Should I add some basic info to the readme?

If you run `npm run makepot`, you do get `genesis-newtheme.pot` (one dash) from the “Genesis NewTheme” name. The issue is that the existing lang file that was renamed during `npm run rename` is named `genesis-new-theme.pot` (two dashes) using a “Genesis NewTheme” name. So when someone runs `npm run makepot`, they'll have two files in the languages folder:

- genesis-newtheme.pot
- genesis-new-theme.pot

There's also an inconsistency between the Theme Name (“Genesis New Theme” instead of the entered “Genesis NewTheme”), and the text domain (“genesis-newtheme”). It likely relates to the UpperCamelCase, so it's an edge case we could log and don't have to solve for immediately.

bharath commented:
@seothemes, @nathanrice 

landing-page.php header contains 'Genesis sample' instead of 'Genesis Sample' and the script is ignoring it instead of renaming.

![Screenshot 2019-07-20 at 2 02 13 PM](https://user-images.githubusercontent.com/41569/61578244-95f27780-ab11-11e9-8cc2-59c479e16c9c.jpg)

Also when I use the new theme string of "BK Child", The theme name is changed to 'B K Child'

![Screenshot 2019-07-20 at 2 02 34 PM](https://user-images.githubusercontent.com/41569/61578345-0f3e9a00-ab13-11e9-8e0f-fb549f3d178f.jpg)

seothemes commented:
Ok, 0.5.0 is ready for testing. I've fixed the file renaming issue - the strings are no longer split by capital letters. Only spaces are used as delimiters now. For example:

The string "BK Theme" will create a `bk-theme.pot` file. "Genesis NewTheme" will create `genesis-newtheme.pot`.

I've also added a new function to rename the actual theme directory. It follows the same rules as above.

@bharath 1 & 3 are fixed with ea5241b and 8ce5656.

bharath commented:
@seothemes 

I am using new theme string of "BK Child" for testing.

1.  Capitalizing theme name string in landing-page.php fixed it.

2. style.css header still says "B K Child" instead of "BK Child"

![Screenshot 2019-07-21 at 12 09 30 PM](https://user-images.githubusercontent.com/41569/61588169-74969780-abb4-11e9-83eb-93a511ddd385.jpg)

and in woocommerce stylesheet comment

![Screenshot 2019-07-21 at 12 09 41 PM](https://user-images.githubusercontent.com/41569/61588174-8415e080-abb4-11e9-9a19-c34650b9fd48.jpg)

3. Renaming genesis-sample-woocommerce.css file to woocommerce.css works and no longer throws an error.

4. Renaming actual theme directory works good.

seothemes commented:
@bharath 2 should be fixed with 0.6.0 ^

bharath commented:
Everything works good. Great work @seothemes

sparkbold commented:
Have these conflicts ever got resolved and merged to current develop? I do need the feature to rename the theme.
